modules/http/src/main/java/com/spotify/ffwd/http/HttpDiscovery.kt

```diff
@@ -0,0 +1,101 @@
+/*-
+ * -\-\-
+ * FastForward HTTP Module
+ * --
+ * Copyright (C) 2016 - 2018 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+
+package com.spotify.ffwd.http
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonSubTypes
+import com.fasterxml.jackson.annotation.JsonSubTypes.Type
+import com.fasterxml.jackson.annotation.JsonTypeInfo
+import com.fasterxml.jackson.annotation.JsonTypeName
+import com.google.common.collect.ImmutableList
+import com.netflix.loadbalancer.LoadBalancerBuilder
+import com.netflix.loadbalancer.Server
+import java.util.*
+import java.util.stream.Collectors
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
+@JsonSubTypes(Type(HttpDiscovery.Static::class), Type(HttpDiscovery.Srv::class))
+interface HttpDiscovery {
+
+    fun apply(
+            builder: LoadBalancerBuilder<Server>, searchDomain: Optional<String>
+    ): LoadBalancerBuilder<Server>
+
+
+    @JsonTypeName("static")
+    data class Static(val servers: List<HostAndPort>) : HttpDiscovery {
+
+        override fun apply(
+                builder: LoadBalancerBuilder<Server>, searchDomain: Optional<String>
+        ): LoadBalancerBuilder<Server> {
+            val servers = this.servers
+                    .stream()
```

Comment:
I think you can remove the stream and collect here, kotlin lets you do functional mappings on the list directly.